test/common/testobjects.go

```diff
@@ -40,7 +40,9 @@ func NewTestObjects(typeConfig typeconfig.Interface, namespace string, clusterNa
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	template.SetNamespace(namespace)
+	if typeConfig.GetNamespaced() {
+		template.SetNamespace(namespace)
```

Comment:
Based on Kubernetes, if the resource is not namespace scoped, then even if we create the resource with namespace, the namespace will have no effect. But +1 to @danehans , it is always good to add those check.

LGTM